org.sofproject.gst.topo/src/org/sofproject/gst/json/JsonCustomOptionPane.java

```diff
@@ -0,0 +1,169 @@
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.swt.widgets.Monitor;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+import org.sofproject.topo.ui.graph.ITopoGraph;
+import org.sofproject.gst.json.JsonUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.sofproject.gst.topo.model.GstTopoGraph;
+
+public class JsonCustomOptionPane {
+
+	ITopoGraph graph;
+
+	private static boolean isInteger(String s) {
+		try {
+			Integer.parseInt(s);
+		} catch (NumberFormatException e) {
+			return false;
+		} catch (NullPointerException e) {
+			return false;
+		}
+		return true;
+	}
+
+	public JsonCustomOptionPane(Display display, ITopoGraph graph) {
+		this.graph = graph;
+		Shell shell = new Shell(display, SWT.CLOSE | SWT.TITLE | SWT.MIN);
+
+		shell.setText("Serialize JSON");
+
+		GridLayout gridLayout = new GridLayout(4, false);
+		gridLayout.verticalSpacing = 8;
+		shell.setLayout(gridLayout);
+
+		new Label(shell, SWT.NULL).setText("Name:");
+		Text nameText = new Text(shell, SWT.SINGLE | SWT.BORDER);
+		GridData nameGridData = new GridData(GridData.HORIZONTAL_ALIGN_FILL);
+		nameGridData.horizontalSpan = 3;
+		nameText.setLayoutData(nameGridData);
+
+		new Label(shell, SWT.NULL).setText("Version:");
+		Text versionText = new Text(shell, SWT.BORDER);
+		GridData versionGridData = new GridData(GridData.HORIZONTAL_ALIGN_FILL);
+		versionGridData.horizontalSpan = 3;
+		versionText.setLayoutData(versionGridData);
+
+		new Label(shell, SWT.NULL).setText("Description:");
+		Text descriptionText = new Text(shell, SWT.BORDER | SWT.MULTI | SWT.V_SCROLL);
+		GridData descriptionGridData = new GridData(GridData.HORIZONTAL_ALIGN_FILL);
+		descriptionGridData.horizontalSpan = 3;
+		descriptionGridData.heightHint = 100;
+		descriptionGridData.widthHint = 200;
+		descriptionText.setLayoutData(descriptionGridData);
+
+		new Label(shell, SWT.NULL).setText("Type:");
+		Combo typeCombo = new Combo(shell, SWT.READ_ONLY);
+		typeCombo.setBounds(50, 50, 200, 65);
+		String items[] = { "Gstreamer", "Ffmpeg" };
+		typeCombo.setItems(items);
+		typeCombo.setText(items[0]);
+		GridData typeGridData = new GridData(GridData.HORIZONTAL_ALIGN_FILL);
+		typeGridData.horizontalSpan = 3;
+		typeCombo.setLayoutData(typeGridData);
+
+		Button okButton = new Button(shell, SWT.PUSH);
+		okButton.setText("Ok");
+		GridData buttonGridData = new GridData(GridData.HORIZONTAL_ALIGN_FILL);
+		okButton.setLayoutData(buttonGridData);
+		okButton.addSelectionListener(new SelectionAdapter() {
```

Comment:
I'm not a swt expert, but this stuff seems too verbose. Are there no more smart ways? Some builders/chaining at least? Also all these variables are introduced for 200 lines scope just to be used within ~5lines.

Reply:
@jajanusz I feel like this is generated code by some graphical tool. I'm not foreign to this kind of stuff even though I myself usually do that with C# rather than Java projects. The uniformity of actual code for each of the elements is a strong hint to me that this is generated code in fact so it doesn't deserve too deep of a review process honestly.

Reply:
Usually IDEs give some marker comments in such case to recognize what's generated/manually added. But it guess different tools can have different ways of handling it.

Reply:
Can live without that change for now.